processor/resourcedetectionprocessor/internal/consul/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consul
+
+import (
+	"go.opentelemetry.io/collector/config"
+)
+
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Address is the address of the Consul server
+	Address string `mapstructure:"address"`
+
+	// Datacenter to use. If not provided, the default agent datacenter is used.
+	Datacenter string `mapstructure:"datacenter"`
+
+	// Token is used to provide a per-request ACL token
+	// which overrides the agent's default token.
+	Token string `mapstructure:"token"`
+
+	// TokenFile is a file containing the current token to use for this client.
```

Comment:
Presumably only one of Token or TokenFile should be set, let's document that and add config.Validate for it

Reply:
conterintuitively, none of these arguments are required and will work for default consul installations. Should I add a comment at the top regarding this?

Reply:
Yeah that'd be helpful